### DISCUSSION TITLE
Pattern for string with size?

### DISCUSSION THREAD

**Discussion by danielniccoli** [WerWolv/ImHex#1937] [Get Help] [answered]
Hi,

I have the following data which translates to a string.

`04 62 61 73 65`

The first byte gives the length of the string and then you have the string.

I have the following pattern that works:

```
u8 len @ 0x0a;
char string[len] @ 0x0b;
```

I wonder if there's maybe a better way to do this?

**Comment by jumanji144** [comment, marked as answer]
Hello, yes there is a better way. Check out the standard library type:  `std::string::SizedString` from the `type.magic` library. In this case you'd use it like this:
```python
import type.magic;

std::string::SizedString<u8> string @ 0x0a;
```
